src/FileSystem/DefaultFileSystem.cpp

```diff
@@ -72,6 +74,22 @@ std::filesystem::path DefaultFileSystem::FindPath(const std::filesystem::path& p
 	throw std::runtime_error("File " + path.string() + " not found");
 }
 
+bool DefaultFileSystem::IsPathValid(const std::filesystem::path& path)
+{
+	if (path.empty())
+		return false;
+
+	std::ifstream file(path);
+
+	if (file.is_open())
+	{
+		file.close();
+		return true;
+	}
```

Comment:
`path` is most likely going to be a directory (though with #674 it will have the option to be a file) so this won't work.

Have a look at [`std::filesystem::exists`](https://en.cppreference.com/w/cpp/filesystem/exists) which is most likely what you want here.

Reply:
@bwrsandman Please see updated fix :)

Reply:
Seems like you're still opening the directory like if it was a file